snuba/subscriptions/executor_consumer.py

```diff
@@ -286,3 +290,74 @@ def join(self, timeout: Optional[float] = None) -> None:
         remaining = timeout - (time.time() - start) if timeout is not None else None
         self.__next_step.close()
         self.__next_step.join(remaining)
+
+
+class ProduceResult(ProcessingStrategy[SubscriptionTaskResult]):
+    """
+    Gets a SubscriptionTaskResult from the ExecuteQuery processor, encodes and produces
+    the results to the subscription result topic then commits offsets.
+    """
+
+    def __init__(
+        self,
+        producer: Producer[KafkaPayload],
+        result_topic: str,
+        commit: Callable[[Mapping[Partition, Position]], None],
+    ):
+        self.__producer = producer
+        self.__result_topic = Topic(result_topic)
+        self.__commit = commit
+
+        self.__encoder = SubscriptionTaskResultEncoder()
+
+        self.__queue: Deque[
+            Tuple[Message[SubscriptionTaskResult], Future[Message[KafkaPayload]]]
+        ] = deque()
+
+        self.__max_buffer_size = 10000
```

Comment:
this is something we will want to tweak. please let's ensure we eventually move it to CLI or settings.

Reply:
👍